In [0]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


#Matrículas

In [0]:

csv_ingressantes2017_1 = 'drive/My Drive/EAD/matricula-componente-20171.csv'
csv_ingressantes2017_2 = 'drive/My Drive/EAD/matricula-componente-20172.csv'
csv_ingressantes2016_1 = 'drive/My Drive/EAD/matricula-componente-20181.csv'
csv_ingressantes2016_2 = 'drive/My Drive/EAD/matricula-componente-20182.csv'
csv_ingressantes2015_1 = 'drive/My Drive/EAD/matricula-componente-20191.csv'


df_ingressantes2017_1 = pd.read_csv(csv_ingressantes2017_1, sep=';')
df_ingressantes2017_2 = pd.read_csv(csv_ingressantes2017_2, sep=';')
df_ingressantes2016_1 = pd.read_csv(csv_ingressantes2016_1, sep=';')
df_ingressantes2016_2 = pd.read_csv(csv_ingressantes2016_2, sep=';')
df_ingressantes2015_1 = pd.read_csv(csv_ingressantes2015_1, sep=';')


In [0]:
matriculas = []

for i in range(15,18):
    for j in range (1,2):
        matriculas.append(vars()['df_ingressantes20%d_%d' % (i, j)])

In [72]:
df_matriculas = pd.concat(matriculas, sort=True)

df_matriculas.reset_index(drop=True,inplace=True)

df_matriculas.tail()

descricao                          discente  ...  reposicao  unidade
1986603   APROVADO  8acac2ab250a3f626d8eae2361d8bce3  ...        NaN      NaN
1986604   APROVADO  1ec032d85a72cd974eee36465309866f  ...        NaN      NaN
1986605   APROVADO  b4e4030b0058e4b41ab8f6edac3bd22d  ...        NaN      NaN
1986606  CANCELADO  36e45a3eba0e103749424284d5684ddf  ...        NaN      NaN
1986607  CANCELADO  c9be47af3402d4c8768da3c89de8f649  ...        NaN      NaN

[5 rows x 10 columns]

In [21]:
matriculas_imd = df_matriculas[df_matriculas.id_curso == 92127264]
matriculas_imd.reset_index(drop=True, inplace=True)
matriculas_imd.head()

descricao                          discente  ...  reposicao  unidade
0          REPROVADO  85a7465d77f7fda11b2cf4afd297df9c  ...      False      2.0
1          REPROVADO  85a7465d77f7fda11b2cf4afd297df9c  ...      False      1.0
2  APROVADO POR NOTA  6056c4082d4a7f65d64943dad44eaac4  ...      False      2.0
3  APROVADO POR NOTA  6056c4082d4a7f65d64943dad44eaac4  ...      False      1.0
4          REPROVADO  2b2e67de30e2dcc7880704f1d6498e70  ...      False      2.0

[5 rows x 10 columns]

In [22]:
matriculas_consolidadas = matriculas_imd[
    (matriculas_imd.descricao != 'EXCLUIDA') & 
    (matriculas_imd.descricao != 'CANCELADO')
]
matriculas_consolidadas.reset_index(drop=True, inplace=True)
matriculas_consolidadas.head()

descricao                          discente  ...  reposicao  unidade
0          REPROVADO  85a7465d77f7fda11b2cf4afd297df9c  ...      False      2.0
1          REPROVADO  85a7465d77f7fda11b2cf4afd297df9c  ...      False      1.0
2  APROVADO POR NOTA  6056c4082d4a7f65d64943dad44eaac4  ...      False      2.0
3  APROVADO POR NOTA  6056c4082d4a7f65d64943dad44eaac4  ...      False      1.0
4          REPROVADO  2b2e67de30e2dcc7880704f1d6498e70  ...      False      2.0

[5 rows x 10 columns]

#Matrículas em Redes

In [74]:
mat_redes = df_matriculas[(df_matriculas.id_turma == 57600983)]
mat_redes

descricao  ... unidade
1622236            APROVADO  ...     2.0
1622237            APROVADO  ...     3.0
1622238            APROVADO  ...     1.0
1622239           CANCELADO  ...     1.0
1622240           CANCELADO  ...     2.0
1622241           CANCELADO  ...     3.0
1622242            APROVADO  ...     1.0
1622243            APROVADO  ...     2.0
1622244            APROVADO  ...     3.0
1622245   APROVADO POR NOTA  ...     1.0
1622246   APROVADO POR NOTA  ...     2.0
1622247   APROVADO POR NOTA  ...     3.0
1938833            APROVADO  ...     2.0
1938834            APROVADO  ...     3.0
1938835            APROVADO  ...     1.0
1938836           REPROVADO  ...     1.0
1938837           REPROVADO  ...     2.0
1938838           REPROVADO  ...     3.0
1938839   APROVADO POR NOTA  ...     1.0
1938840   APROVADO POR NOTA  ...     2.0
1938841   APROVADO POR NOTA  ...     3.0
1938842            APROVADO  ...     3.0
1938843            APROVADO  ...     1.0
1938844            APROVADO  ...     2.0
1938845            APROVADO  ...     1.0
1938846            APROVADO  ...     2.0
1938847            APROVADO  ...     3.0
1938848            APROVADO  ...     1.0
1938849            APROVADO  ...     2.0
1938850            APROVADO  ...     3.0
1938851            APROVADO  ...     2.0
1938852            APROVADO  ...     1.0
1938853            APROVADO  ...     3.0
1938854            APROVADO  ...     1.0
1938855            APROVADO  ...     2.0
1938856            APROVADO  ...     3.0
1938857  REPROVADO POR NOTA  ...     1.0
1938858  REPROVADO POR NOTA  ...     3.0
1938859  REPROVADO POR NOTA  ...     2.0

[39 rows x 10 columns]

#Ingressantes

In [0]:
csv_ingressantes2017_1 = 'drive/My Drive/EAD/turmas-2017.1.csv'
csv_ingressantes2017_2 = 'drive/My Drive/EAD/turmas-2017.2.csv'
csv_ingressantes2016_1 = 'drive/My Drive/EAD/turmas-2016.1.csv'
csv_ingressantes2016_2 = 'drive/My Drive/EAD/turmas-2018.2.csv'
csv_ingressantes2015_1 = 'drive/My Drive/EAD/turmas-2018.1.csv'


df_ingressantes2017_1 = pd.read_csv(csv_ingressantes2017_1, sep=';')
df_ingressantes2017_2 = pd.read_csv(csv_ingressantes2017_2, sep=';')
df_ingressantes2016_1 = pd.read_csv(csv_ingressantes2016_1, sep=';')
df_ingressantes2016_2 = pd.read_csv(csv_ingressantes2016_2, sep=';')
df_ingressantes2015_1 = pd.read_csv(csv_ingressantes2015_1, sep=';')

In [0]:
ingressantes = []

for i in range(15,18):
    for j in range (1,2):
        ingressantes.append(vars()['df_ingressantes20%d_%d' % (i, j)])

In [25]:
df_ingressantes = pd.concat(ingressantes)

df_ingressantes.reset_index(drop=True,inplace=True)

df_ingressantes.tail()

id_turma codigo_turma  ...  convenio  modalidade_participantes
43396  57612971           01  ...       NaN                Presencial
43397  57613796           01  ...       NaN               A Distância
43398  57613796           01  ...       NaN               A Distância
43399  57613798           02  ...       NaN               A Distância
43400  57613798           02  ...       NaN               A Distância

[5 rows x 26 columns]

#Catando o Componente

In [0]:
componentes_presenciais = 'drive/My Drive/EAD/componentes-curriculares-presenciais.csv'


componentes = pd.read_csv(componentes_presenciais, sep=';')

In [40]:
componentes.head()

id_componente tipo_componente  ...  modalidade curso_componente
0          18642      DISCIPLINA  ...  Presencial              NaN
1          18643      DISCIPLINA  ...  Presencial              NaN
2          18644      DISCIPLINA  ...  Presencial              NaN
3          18645      DISCIPLINA  ...  Presencial              NaN
4          18646      DISCIPLINA  ...  Presencial              NaN

[5 rows x 29 columns]

###Segurança de Redes de Computadores

In [41]:
cod_comp = 'DCA0402'

componentes_filtrado = componentes[componentes.codigo == cod_comp]
componentes_filtrado.head()

id_componente tipo_componente  ...  modalidade curso_componente
10833          40127      DISCIPLINA  ...  Presencial              NaN

[1 rows x 29 columns]

###Segurança de Redes (**IMD**)

In [42]:
cod_comp = 'IMD0703'

componentes_filtrado = componentes[componentes.codigo == cod_comp]
componentes_filtrado.head()

id_componente tipo_componente  ...  modalidade curso_componente
26950          62887          MÓDULO  ...  Presencial              NaN

[1 rows x 29 columns]

# Tentando encontrar a Turma

In [51]:
df_ingressantes.nivel_ensino.unique()

array(['LATO SENSU', 'TÉCNICO', 'STRICTO SENSU', 'GRADUAÇÃO',
       'FORMAÇÃO COMPLEMENTAR', 'RESIDÊNCIA'], dtype=object)

In [52]:
graduacao = df_ingressantes[df_ingressantes.nivel_ensino == 'GRADUAÇÃO']
graduacao.reset_index(drop=True, inplace=True)
graduacao.head()

id_turma codigo_turma  ...  convenio  modalidade_participantes
0  57613134           01  ...       NaN                Presencial
1  57613135           01  ...       NaN                Presencial
2  57613136           01  ...       NaN                Presencial
3  57613136           01  ...       NaN                Presencial
4  57613137           01  ...       NaN                Presencial

[5 rows x 26 columns]

In [53]:
graduacao.tipo.unique()

array(['REGULAR', 'ENSINO INDIVIDUAL', 'REPOSIÇÃO NO PERÍODO REGULAR'],
      dtype=object)

In [54]:
turmas = graduacao[graduacao.tipo == 'REGULAR']
turmas.reset_index(drop=True, inplace=True)
turmas.head()

id_turma codigo_turma  ...  convenio  modalidade_participantes
0  57613134           01  ...       NaN                Presencial
1  57613135           01  ...       NaN                Presencial
2  57613136           01  ...       NaN                Presencial
3  57613136           01  ...       NaN                Presencial
4  57613137           01  ...       NaN                Presencial

[5 rows x 26 columns]

In [55]:
turmas.situacao_turma.unique()

array(['CONSOLIDADA', 'EXCLUÍDA', 'INTERROMPIDA', 'ABERTA'], dtype=object)

In [56]:
turmas_consolidadas = turmas[turmas.situacao_turma == 'CONSOLIDADA']
turmas_consolidadas.reset_index(drop=True, inplace=True)
turmas_consolidadas.head()

id_turma codigo_turma  ...  convenio  modalidade_participantes
0  57613134           01  ...       NaN                Presencial
1  57613135           01  ...       NaN                Presencial
2  57613136           01  ...       NaN                Presencial
3  57613136           01  ...       NaN                Presencial
4  57613139           01  ...       NaN                Presencial

[5 rows x 26 columns]

In [57]:
id_comp = 40127 #Segurança de redes

filtro_ingressantes = df_ingressantes[df_ingressantes.id_componente_curricular == id_comp]
filtro_ingressantes.head()

id_turma codigo_turma  ...  convenio  modalidade_participantes
40925  57600983           01  ...       NaN                Presencial

[1 rows x 26 columns]

#Tentando filtrar o componente - **Fail for 62887 (IMD0703)**

In [37]:
id_componente = 40127

turmas_filtradas = turmas[turmas.id_componente_curricular == id_componente]
turmas_filtradas = turmas_filtradas.filter(items=['id_turma', 'id_componente_curricular', 'ano', 'id_docente_interno'])
turmas_filtradas.head()

id_turma  id_componente_curricular   ano
32912  57600983                     40127  2017

#Docentes

In [0]:
csv_docentes = 'drive/My Drive/EAD/docentes.csv'

df_docentes = pd.read_csv(csv_docentes, sep=';')

In [44]:
df_docentes.reset_index(drop=True,inplace=True)

df_docentes.tail()

siape  ...                       admissao
2365   346866  ...  1977/03/01 00:00:00.000000000
2366  2374871  ...  2013/02/18 00:00:00.000000000
2367   350630  ...  1989/01/26 00:00:00.000000000
2368  2177287  ...  2004/07/14 00:00:00.000000000
2369  1753896  ...  2010/01/26 00:00:00.000000000

[5 rows x 11 columns]

In [45]:
print(df_docentes.categoria.unique())

['PROFESSOR DE 1  E 2  GRAUS              '
 'PROFESSOR DE ENSINO BASICO TECNICO E TECNOLOGICO'
 'PROFESSOR DO ENSINO BASICO, TECNICO E TECNOLOGICO'
 'PROFESSOR DO MAGISTERIO SUPERIOR']


In [46]:
docentes_graduacao = df_docentes[df_docentes.categoria == 'PROFESSOR DO MAGISTERIO SUPERIOR']
docentes_graduacao.reset_index(drop=True, inplace=True)
docentes_graduacao.head()

siape  ...                       admissao
0   345783  ...  1973/03/01 00:00:00.000000000
1  1668850  ...  2016/04/08 00:00:00.000000000
2  1222082  ...  1997/04/22 00:00:00.000000000
3  6345784  ...  2011/12/01 00:00:00.000000000
4  2166798  ...  2014/10/02 00:00:00.000000000

[5 rows x 11 columns]

In [47]:
docentes_graduacao.lotacao.unique()

array(['DEPARTAMENTO DE ENGENHARIA ELÉTRICA',
       'CERES - DEPARTAMENTO DE HISTÓRIA',
       'DEPARTAMENTO DE ENGENHARIA CIVIL',
       'DEPARTAMENTO DE ENGENHARIA DE COMUNICACOES',
       'ESCOLA MULTICAMPI DE CIÊNCIAS MÉDICAS DO RIO GRANDE DO NORTE',
       'ESCOLA DE CIÊNCIAS E TECNOLOGIA', 'ESCOLA DE MÚSICA',
       'DEPARTAMENTO DE ENGENHARIA DE COMPUTACAO E AUTOMACAO',
       'INSTITUTO DE QUÍMICA', 'DEPARTAMENTO DE GEOFÍSICA',
       'DEPARTAMENTO DE CIÊNCIAS CONTÁBEIS - DCC',
       'DEPARTAMENTO DE ENGENHARIA MECANICA',
       'DEPARTAMENTO DE FUNDAMENTOS E POLÍTICAS DA EDUCAÇÃO',
       'INSTITUTO METROPOLE DIGITAL', 'DEPARTAMENTO DE FARMACIA',
       'DEPARTAMENTO DE ANÁLISES CLÍNICAS E TOXICOLÓGICAS',
       'DEPARTAMENTO DE ODONTOLOGIA',
       'FACULDADE DE CIÊNCIAS DA SAÚDE DO TRAIRI - FACISA',
       'DEPARTAMENTO DE BIOLOGIA CELULAR E GENÉTICA',
       'DEPARTAMENTO DE LINGUAS E LITERATURAS ESTRANGEIRAS MODERNAS',
       'DEPARTAMENTO DE ECOLOGIA', 'DEPARTAMENTO DE 

In [48]:
docentes_imd = docentes_graduacao[(docentes_graduacao.lotacao == 'DEPARTAMENTO DE INFORMÁTICA E MATEMÁTICA APLICADA') | (docentes_graduacao.lotacao ==  'INSTITUTO METROPOLE DIGITAL')]
docentes_imd.reset_index(drop=True, inplace=True)
docentes_imd.head()

siape  ...                       admissao
0  2451906  ...  2012/11/01 00:00:00.000000000
1  2645969  ...  2013/04/25 00:00:00.000000000
2  2278573  ...  2016/01/19 00:00:00.000000000
3  1363515  ...  2002/10/08 00:00:00.000000000
4  1885001  ...  2017/05/26 00:00:00.000000000

[5 rows x 11 columns]

#Filtrando Professores

In [67]:
turmas_redes.id_turma

32912    57600983
Name: id_turma, dtype: int64

In [75]:
turmas_redes = turmas_filtradas
turmas_redes['aprovacao'] = 'NaN'


for index, row in turmas_redes.iterrows():
    filtro = df_matriculas[
        df_matriculas['id_turma'] == row['id_turma']
    ]
    
    turmas_redes.loc[index, 'aprovacao'] = (filtro[
        filtro['descricao'].isin(['APROVADO', 'APROVADO POR NOTA'])
    ]['discente'].count() * 100 / 
        filtro['discente'].count()
    )

turmas_redes

id_turma  id_componente_curricular   ano aprovacao
32912  57600983                     40127  2017   76.9231

#Aprovação

In [76]:

turmas_redes['aprovacao'] = turmas_redes['aprovacao'].astype(float)
turmas_redes['aprovacao'].mean()

76.92307692307692

#Conclusão: 
A disciplina IMD0703 - Segurança de Redes foi cadastrada no programa em 2019 e portanto não existem dados suficientes na base de dados para serem analisados.
No entanto, por fins demonstrativos, realizamos a análise baseada nos dados referentes a disciplina Segurança de Redes de Computadores - DCA0204, ofertada no departamento de Engenharia da computação.
A análise trouxe como resultado uma taxa de aprovação geral de cerca de 76.9% para a disciplina.